saliency/base.py

```diff
@@ -95,3 +98,83 @@ def GetMask(self, x_value, feed_dict={}):
     """
     feed_dict[self.x] = [x_value]
     return self.session.run(self.gradients_node, feed_dict=feed_dict)[0]
+
+
+OUTPUT_GRADIENTS = 'OUTPUT_GRADIENTS'
+CONVOLUTION_LAYER = 'CONVOLUTION_LAYER'
+CONVOLUTION_GRADIENTS = 'CONVOLUTION_GRADIENTUS'
+
+
+class CallModelSaliency(object):
+  r"""Base class for saliency masks. Alone, this class doesn't do anything."""
+
+  def __init__(self):
+    pass
+
+  def GetMask(self, x_value, call_model_function, call_model_args=None):
+    """Returns an unsmoothed mask.
+
+    Args:
+      x_value: Input ndarray.
+      call_model_function: A function that interfaces with a model to return
+        specific data in a dictionary when given an input and other arguments.
+        Expected function signature:
+        - call_model_function(x_value_batch,
+                              call_model_args=None,
+                              expected_keys=None):
+          x_value_batch - Input for the model, given as a batch (i.e. dimension
+            0 is the batch dimension, dimensions 1 through n represent a single
+            input).
+          call_model_args - Other arguments used to call and run the model.
+          expected_keys - List of keys that are expected in the output.
+      call_model_args: The arguments that will be passed to the call model
+        function, for every call of the model.
+
+    """
+    raise NotImplementedError('A derived class should implemented GetMask()')
+
+  def GetSmoothedMask(self,
+                      x_value,
+                      call_model_function,
+                      call_model_args=None,
+                      stdev_spread=.15,
+                      nsamples=25,
+                      magnitude=True,
+                      **kwargs):
```

Comment:
batch_size?

Reply:
Unsure if batch_size is always going to work with all CallModelSaliency methods. For example, GradCam is a single calculation of gradients, so what would the batch_size do there?